runtime/onert/api/python/package/experimental/train/dataloader.py

```diff
@@ -126,22 +139,22 @@ def _load_raw(self, file_path, shape, dtype):
         # Read and load the raw data
         with open(file_path, "rb") as f:
             data = f.read()
-            array = np.frombuffer(data, dtype=dtype)
-            if array.size != expected_elements:
-                raise ValueError(
-                    f"Raw data size does not match the expected shape: {shape}. "
-                    f"Expected {expected_elements} elements, got {array.size} elements.")
-            return array.reshape(shape)
+        array = np.frombuffer(data, dtype=dtype)
+        if array.size != expected_elements:
+            raise ValueError(
+                f"Raw data size does not match the expected shape: {shape}. "
+                f"Expected {expected_elements} elements, got {array.size} elements.")
+        return array.reshape(shape)
```

Comment:
I only keep `f.read() `in the `with `so the file is open just to load the bytes. Once data is in memory, `np.frombuffer`, size checks, and reshape work on that buffer(no open file needed) so they live outside the `with`.